### PR TITLE
Add dynamic commitment

### PR DESCRIPTION
**Pull request by lukegravitt**
- Update help messages for all assignment options
- Rename Edit to Modify
- Update suspend icon
- Add action for new dynamic commit
- Fix popup-proxy background color on mobile

### CODE REVIEW

File: src/pages/assignments/components/assignment-card.vue [hunk 1]
```diff
@@ -214,13 +224,57 @@ q-card.assignment(v-if="isFiltered && ((isExpired && history) || (!isExpired &&
         )
           q-item-section(style="max-width: 20px;")
             q-icon(name="fas fa-pencil-alt" size="14px")
-          q-item-section Edit
+          q-item-section Modify
+        q-item(
+          v-if="account === assignee"
+          clickable
+        )
+          q-popup-proxy(@before-show="newCommit = commited")
```
Comment (gregory-latinier) [inline src/pages/assignments/components/assignment-card.vue:232]: Committed no? with 2 T ?

File: src/pages/assignments/components/assignment-card.vue [hunk 2]
```diff
@@ -214,13 +224,57 @@ q-card.assignment(v-if="isFiltered && ((isExpired && history) || (!isExpired &&
...
+            .confirm.column.q-pa-sm(style="max-width: 275px; background:white;")
```
Comment (gregory-latinier) [inline src/pages/assignments/components/assignment-card.vue:233]: There's a class helper for color `bg-white`

File: src/pages/assignments/components/assignment-card.vue [hunk 3]
```diff
@@ -214,13 +224,57 @@ q-card.assignment(v-if="isFiltered && ((isExpired && history) || (!isExpired &&
...
+              | If you adjust your assignment, your % commitment will be immediately
+              | reflected on your next claim (no vote is necessary). Multiple adjustments
+              | during the same claim period will be included in the calculation.
+              .commit-group.q-mt-sm.q-px-sm.q-pb-sm(style="background: #EFEFEF;")
```
Comment (gregory-latinier) [inline src/pages/assignments/components/assignment-card.vue:237]: You create a new css class but not use it and add inline CSS
I would use the css class you created

File: src/pages/assignments/components/assignment-card.vue [hunk 4]
```diff
@@ -214,13 +224,57 @@ q-card.assignment(v-if="isFiltered && ((isExpired && history) || (!isExpired &&
...
+                .row.q-pt-xs.q-pb-md New % Commitment
+                .row.q-mt-sm.q-px-md
+                  q-slider(
+                    v-model="newCommit"
+                    :min="0"
+                    :max="commited"
```
Comment (gregory-latinier) [inline src/pages/assignments/components/assignment-card.vue:243]: committed?

File: src/pages/assignments/components/assignment-card.vue [hunk 5]
```diff
@@ -214,13 +224,57 @@ q-card.assignment(v-if="isFiltered && ((isExpired && history) || (!isExpired &&
...
+                .row.text-caption.justify-between
+                  span 0%
+                  span {{ commited + '%' }}
+              .row.flex.justify-between.q-mt-sm
+                q-btn(
+                  color="primary"
+                  label="No"
+                  dense
+                  flat
+                  v-close-popup="-1"
```
Comment (gregory-latinier) [inline src/pages/assignments/components/assignment-card.vue:259]: v-close-popup is enough I think
Is there any reason why you put `="-1"` ?

File: src/pages/assignments/components/assignment-form.vue [hunk 6]
```diff
@@ -161,6 +161,7 @@ export default {
   strong.title
     span {{ title }}
     span(v-if="form.edit") &nbsp; (Extending)
+  .edit-note(v-if="form.edit") If you modify your assignment you will propose your edits in a new vote. Once passed, your edits will merge with the existing assignment.
```
Comment (gregory-latinier) [inline src/pages/assignments/components/assignment-form.vue:164]: Why add a class if you're not adding css?

**Review by gregory-latinier — CHANGES_REQUESTED** [review]
Good work :)

File: src/pages/assignments/components/assignment-card.vue (hunk 5 above)
Comment (lukegravitt) [reply src/pages/assignments/components/assignment-card.vue:259]: This was copied from other existing entries in the file. Quasar says it is needed to close a chain of popups and in this is situation, I think that's correct.

https://quasar.dev/vue-directives/close-popup#Closing-multiple-levels

Comment (gregory-latinier) [reply src/pages/assignments/components/assignment-card.vue:259]: Ok I see I forgot about this

**Comment by lukegravitt** [comment]
This should be working. In doing this I discovered bugs in both suspend/withdraw: (1) using an id field that doesn't seem to exist anymore -> switched it to hash and (2) fixed the redirect after those are called so that it doesnt 404 (there was an extra 's'). However, neither of those actions seems to be implemented on the backend at the moment from my perspective, so we will probably need to do a bit more testing on all these functions as those features are brought in

**Comment by lukegravitt** [comment]
Ready for review!

File: src/store/assignments/actions.js [hunk 7]
```diff
@@ -148,28 +157,46 @@ export const claimAssignmentPayment = async function (context, hash) {
...
+    }
+  }]
+
+  return this.$api.signTransaction(actions)
+}
+
+export const suspendAssignment = async function ({ rootState }, hash) {
   const actions = [{
     account: this.$config.contracts.dao,
     name: 'propsuspend',
```
Comment (gregory-latinier) [inline src/store/assignments/actions.js:181]: This action doesn't exist anymore apparently

File: src/pages/assignments/components/assignment-card.vue [hunk 8]
```diff
@@ -159,6 +163,16 @@ export default {
     roleId () {
       return this.getValue(this.assignment, 'details', 'role')
     },
+    committed () {
+      return this.getValue(this.assignment, 'details', 'time_share_x100')
+    },
+    isAdjusted () {
+      if (this.assignment && this.assignment.lastimeshare) {
+        let timeShare = this.getValue(this.assignment.lastimeshare[0], 'details', 'time_share_x100')
+        return timeShare < this.committed
```
[the remark this answers is not in the document]
Comment (lukegravitt) [reply src/pages/assignments/components/assignment-card.vue:172]: This is only used for the UX (shows when you reduced below the role max). Check the details in #203. I'll adjust the variable names a bit (committed is really something like assignment-initial-commit [which is used as the max]).

File: src/store/assignments/actions.js (hunk 7 above)
Comment (gregory-latinier) [reply src/store/assignments/actions.js:181]: OK let's skip this then

File: src/store/assignments/actions.js [hunk 9]
```diff
@@ -148,28 +157,46 @@ export const claimAssignmentPayment = async function (context, hash) {
...
   }]
 
   return this.$api.signTransaction(actions)
 }
 
-export const withdrawFromAssignment = async function ({ rootState }, { id, notes }) {
+export const withdrawFromAssignment = async function ({ rootState }, { hash, notes }) {
   const actions = [{
     account: this.$config.contracts.dao,
     name: 'withdraw',
```
[the remark this answers is not in the document]
Comment (lukegravitt) [reply src/store/assignments/actions.js:195]: Same as above. Let me know if you want me to comment these out as part of this pr.

**Comment by lukegravitt** [comment]
@gregory-latinier Updated variable name for clarity. Can't merge until the block is resolved. Thanks!

**Review by gregory-latinier — APPROVED** [review]

**Pull request merged**